localstack/services/transcribe/provider.py

```diff
@@ -0,0 +1,285 @@
+            summaries.append(
+                TranscriptionJobSummary(
+                    TranscriptionJobName=job["TranscriptionJobName"],
+                    LanguageCode=job["LanguageCode"],
+                    CreationTime=job["CreationTime"],
+                    StartTime=job["StartTime"],
+                    TranscriptionJobStatus=job["TranscriptionJobStatus"],
+                    CompletionTime=job.get("CompletionTime"),
+                    FailureReason=job.get("FailureReason"),
+                )
+            )
+
+        return ListTranscriptionJobsResponse(TranscriptionJobSummaries=summaries)
+
+    def delete_transcription_job(
+        self, context: RequestContext, transcription_job_name: TranscriptionJobName
+    ) -> None:
+        store = transcribe_stores[context.account_id][context.region]
+
+        if transcription_job_name not in store.transcription_jobs:
+            raise NotFoundException(
+                "The requested job couldn't be found. Check the job name and try your request again."
+            )
+
+        store.transcription_jobs.pop(transcription_job_name)
+
+    #
+    # Utils
+    #
+
+    @staticmethod
+    def download_model(name: str):
+        """
+        Download a Vosk language model to LocalStack cache directory. Do nothing if model is already downloaded.
+
+        While can Vosk also download a model if not available locally, it saves it to a
+        non-configurable location ~/.cache/vosk.
+        """
+        model_path = LANGUAGE_MODEL_DIR / name
+
+        with _DL_LOCK:
+            if (model_path).exists():
+                return
+            else:
+                model_path.mkdir(parents=True)
+
+            model_zip_path = str(model_path) + ".zip"
+
+            LOG.debug("Downloading language model: %s", model_path.name)
+            download(MODEL_PRE_URL + str(model_path.name) + ".zip", model_zip_path)
+
+            LOG.debug("Extracting language model: %s", model_path.name)
+            with ZipFile(model_zip_path, "r") as model_ref:
+                model_ref.extractall(model_path.parent)
+
+            Path(model_zip_path).unlink()
+
+    #
+    # Threads
+    #
+
+    def _run_transcription_job(self, args: Tuple[TranscribeStore, str]):
+        store, job_name = args
+
+        job = store.transcription_jobs[job_name]
+        job["StartTime"] = datetime.datetime.utcnow()
+        job["TranscriptionJobStatus"] = TranscriptionJobStatus.IN_PROGRESS
+
+        failure_reason = None
+
+        try:
+            LOG.debug("Starting transcription: %s", job_name)
+
+            # Get file from S3
+            file_path = new_tmp_file()
+            s3_client = aws_stack.connect_to_service("s3")
+            s3_path = job["Media"]["MediaFileUri"]
+            bucket, _, key = s3_path.removeprefix("s3://").partition("/")
+            content = s3_client.get_object(Bucket=bucket, Key=key)["Body"].read()
+            save_file(file_path, content)
```

Comment:
nit: maybe it would be better to use the `download_file` method of the client, which will use streaming IO to write to disk, instead of reading everything into memory first https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-example-download-file.html

Reply:
Great idea, changed in b43741d